src/EnergyPlus/WaterManager.cc

```diff
@@ -950,8 +956,32 @@ namespace WaterManager {
                 ScaleFactor = 0.0;
             }
             state.dataWaterData->RainFall.CurrentRate = schedRate * ScaleFactor / DataGlobalConstants::SecInHour; // convert to m/s
-            state.dataWaterData->RainFall.CurrentAmount = state.dataWaterData->RainFall.CurrentRate * (TimeStepSys * DataGlobalConstants::SecInHour);
+        } else {
+            // placeholder: add EP checks for out of range precipitation value later -- yujie
+            // when there's no site:precipitation but non-zero epw precipitation, uset the epw precipitation as the CurrentRate
+            if (state.dataEnvrn->LiquidPrecipitation > 0.0) {
+                // LiquidPrecipitation is for a certain timestep in an hour, the rate = depth / seconds in a timestep
+                state.dataWaterData->RainFall.CurrentRate =
+                    state.dataEnvrn->LiquidPrecipitation / (DataGlobalConstants::SecInHour / state.dataGlobal->NumOfTimeStepInHour);
+            } else {
+                state.dataWaterData->RainFall.CurrentRate = 0.0;
+            }
+        }
+        state.dataWaterData->RainFall.CurrentAmount = state.dataWaterData->RainFall.CurrentRate * (DataGlobalConstants::SecInHour / state.dataGlobal->NumOfTimeStepInHour);
+        // follow develop branch behavior
+        if (state.dataGlobal->TimeStep > 1) {
+            state.dataEcoRoofMgr->CurrentPrecipitation = state.dataWaterData->RainFall.CurrentAmount; //  units of m
+        }
+        int month = state.dataEnvrn->Month;
+        // change unit back to mm in the reporting in monthly rain amount used in rain collector
+        if ((state.dataWaterData->RainFall.CurrentAmount > 0.0) && (state.dataEnvrn->RunPeriodEnvironment)) {
```

Comment:
Seems like this belongs in `CalcRainCollector`. And should be `if (!state.dataGlobal->WarmupFlag)` same as below. Or both should include `&& RunPeriodEnviornment` ?? I'm not seeing why these ifs should be different from each other.

Reply:
I moved these accumulators to the corresponding CalcRainCollector and UpdateSoilProps.

The ifs should be the same. I changed them both to RunPeriodEnvironment.